conformance/lang/expressions/list-constructor-expr/spread_operator.balt

```diff
@@ -0,0 +1,728 @@
+    Foo[] v2 = [...a1];
+    io:println(v0); // @output []
+    io:println(v1); // @output [{"i":10},{"i":15}]
+    io:println(v2); // @output [{"i":10},{"i":15}]
+}
+
+Test-Case: output
+Description: Test spread operator with union of record types.
+Labels: array-type, list-constructor-expr, spread-operator, tuple-type, mapping-constructor-expr, module-type-defn, record-type,
+        union-type
+
+type Foo record {|int i;|};
+type Bar string|int|Foo;
+
+function init() {
+    (Bar|Foo)[] a0 = [];
+    (Bar|Foo)[4] a1 = [{i:10}, {i:15}, 1, 2];
+    (Bar|Foo)[] v0 = [...a0];
+    [Bar|Foo, Bar|Foo, Bar|Foo, Bar|Foo] v1 = [...a1];
+    (Bar|Foo)[] v2 = [...a1];
+    io:println(v0); // @output []
+    io:println(v1); // @output [{"i":10},{"i":15},1,2]
+    io:println(v2); // @output [{"i":10},{"i":15},1,2]
+}
+
+Test-Case: output
+Description: Test multiple spread operators with list constructor expr.
+Labels: array-type, list-constructor-expr, spread-operator, var
+
+function init() {
+    int[] a0 = [];
+    int[2] a1 = [];
+    int[2] a2 = [1, 2];
+    int[] a3 = [2, 2, 2];
+    int[] v0 = [...a0, ...a1];
+    var v1 = [...a1, ...a2];
+    int[] v2 = [...a2, ...a1];
+    int[] v3 = [...a0, ...a2, ...a1];
+    int[] v4 = [...a0, ...a2, ...a1, ...a3];
+    io:println(v0); // @output [0,0]
+    io:println(v1); // @output [0,0,1,2]
+    io:println(v2); // @output [1,2,0,0]
+    io:println(v3); // @output [1,2,0,0]
+    io:println(v4); // @output [1,2,0,0,2,2,2]
+}
+
+Test-Case: output
+Description: Test spread operator when there is union contextually expected type.
+Labels: array-type, list-constructor-expr, spread-operator, union-type
+
+function init() {
+    int[] a0 = [];
+    int[2] a1 = [];
+    int[2] a2 = [1, 2];
+    int[] a3 = [2, 2, 2];
+    int[] v0 = [...a0, ...a1];
+    (int|string)[] v1 = [...a1, ...a2];
+    (int|string)[] v2 = [...a2, ...a1];
+    (int|string)[] v3 = [...a0, ...a2, ...a1];
+    (int|string)[] v4 = [...a0, ...a2, ...a1, ...a3];
+    io:println(v0); // @output [0,0]
+    io:println(v1); // @output [0,0,1,2]
+    io:println(v2); // @output [1,2,0,0]
+    io:println(v3); // @output [1,2,0,0]
+    io:println(v4); // @output [1,2,0,0,2,2,2]
+}
+
+Test-Case: output
+Description: Test spread operator when there is union contextually expected type.
+Labels: array-type, list-constructor-expr, spread-operator, union-type
+
+function init() {
+    (int|string)[] a0 = [];
+    (int|string)[2] a1 = [0, 0];
+    (int|string)[2] a2 = [1, 2];
+    (int|string)[] a3 = [2, 2, 2];
+    (int|string)[] v0 = [...a0, ...a1];
+    (int|string)[] v1 = [...a1, ...a2];
+    (int|string)[] v2 = [...a2, ...a1];
+    (int|string)[] v3 = [...a0, ...a2, ...a1];
+    (int|string)[] v4 = [...a0, ...a2, ...a1, ...a3];
+    io:println(v0); // @output [0,0]
+    io:println(v1); // @output [0,0,1,2]
+    io:println(v2); // @output [1,2,0,0]
+    io:println(v3); // @output [1,2,0,0]
+    io:println(v4); // @output [1,2,0,0,2,2,2]
+}
+
```

Comment:
We need a test to cover 

> If there is a spread-list-member, and the type of its expression is not fixed length, then the minimum of the total number of members that may be added to the list being constructed by this spread-list-member and any previous list-member must be greater than or equal to the minimum length required by the inherent type

Reply:
It seems the spec has been updated with the spread operator changes recently. 
Shall we check and add more tests accordingly?